Combobox implementation

zer0code9 opened the pull request:
changed the selection from select component to combobox nested component, it uses command and popover
-they looks like the previous select components
-works on phone
-added a system that blocks a combobox from being opened until the previous has its data selected (not for the semester selection)

zer0code9 committed:
Introduced combox to replace select

zer0code9 committed:
Merge branch 'fix/builds' into combobox-implementation

zer0code9 committed:
better combobox

-fixed size and padding
-added icons
-made it so it can only be used if the previous's data was selected

YuraT commented on package.json:
these new deps can be in devDependencies since they're only needed for the build

YuraT commented on package-lock.json:
get rid of package-lock.json, since bun should be used for package management

YuraT commented on src/components/search.tsx:
looks good

zer0code9 committed:
yura fix and style fix for smart search

-updated package.json, and deleted package-lock.json?
-made the selection better with a disabled state
-changed the contents of the smart search so that it is similar to the combobox

YuraT reviewed (approved):
:+1: looks good

joshua-demo reviewed (commented):
Can you please add a video or a screenshot of the rendered page?

zer0code9 commented:
The program will disable the next selection until the current one is chosen. If a selection is changed, the next(s) will reset if applicable. It also works properly on a phone. I'll add the changes made to the dev branch to this branch.

https://github.com/user-attachments/assets/d6cca0fd-345a-4855-acaf-af01a688c41c

zer0code9 committed:
Merge branch 'dev' into combobox-implementation to get new changes

zer0code9 committed:
Merge branch 'dev' into combobox-implementation
